File: conformance/cases/Maps/maps.go
package maps

type Item struct {
	Sku string `json:"sku"`
	Qty int64  `json:"qty"`
}

// Maps exercises typed map values: a scalar-valued and an object-valued map
// must produce hash-identical IR across languages, value type and all.
type Maps struct {
	Labels map[string]string `json:"labels"`
	Items  map[string]Item   `json:"items"`
}

File: conformance/cases/Maps/Maps.java
package conformance;

import jakarta.annotation.Nonnull;
import java.util.Map;

public class Maps {
    @Nonnull public Map<String, String> labels;
    @Nonnull public Map<String, Item> items;

    public static class Item {
        @Nonnull public String sku;
        public long qty;
    }
}
